parsl/tests/test_monitoring/test_exit_helper.py

```diff
@@ -0,0 +1,56 @@
+import multiprocessing
+import signal
+
+import psutil
+import pytest
+
+from parsl.monitoring.monitoring import join_terminate_close_proc
+from parsl.multiprocessing import ForkProcess
+
+
+def noop():
+    pass
+
+
+@pytest.mark.local
+def test_pre_exit():
+    """Test calling against a process that has already exited."""
+    p = ForkProcess(target=noop)
+    p.start()
+    p.join()
+    join_terminate_close_proc(p)
+
+
+def hang():
+    while True:
+        pass
+
+
+@pytest.mark.local
+def test_hang():
+    """Test calling against a process that will not exit itself."""
+    p = ForkProcess(target=hang)
+    p.start()
+    pid = p.pid
+    join_terminate_close_proc(p, timeout=1)
+    assert not psutil.pid_exists(pid), "process should not exist any more"
```

Comment:
If you don't want to use `psutil` ("another lib"), I believe the ForkProcess class can do equivalently via:
```python
assert p.exitcode is not None, ... # one option
assert not p.is_alive(), ... # another
```
Similarly, the return code should be the negation of the signal, so could ensure that it was closed by the signal (as opposed to crashing for, say, a SIGBUS :wink:) via:
```python
assert p.exitcode == -signal.SIGTERM, ...
```

Reply:
not after the process object has been closed by join_terminate_close_proc! that was my first implementation.

Reply:
Ah; the `.close()`.  Hmm; I wonder if `.join_()` should consider returning anything then?  Like the exit code?  Not a big detail, but "perhaps."